src/main/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStep.java

```diff
@@ -110,19 +110,23 @@ public final void checkout(Run<?,?> run, FilePath workspace, TaskListener listen
                     changelogFile = Files.createTempFile(run.getRootDir().toPath(), "changelog", ".xml").toFile();
                 }
             }
+            long changelogOriginalModifiedDate = (changelogFile != null) ? changelogFile.lastModified() : 1L;
```

Comment:
I wouldn't use `1L` as a default here even if it should never be a real value in practice. I would declare the variable as a capital-L `Long` and set it to `null` when `changelogFile == null`.

Reply:
Thought I had it.
```java
Long changelogOriginalModifiedDate = (changelogFile != null) ? changelogFile.lastModified() : null;
```
I misunderstood your suggestion, and was instead putting `Long changelogOriginalModifiedDate = null;` just above the beginning of the `try...` block. Because that was how my brain interpreted "...when `changelogFile == null`..." in your comment. I tried putting it in all manner of strange places and was still like "this just doesn't work."

Even so, these two from workflow-job are still throwing an NPE. So I've still got some figuring to do.

I blame it on four hours of sleep, coupled with a 6yo who asks me if I can come play with her every 10 minutes.

Reply:
> ```
> Long changelogOriginalModifiedDate = (changelogFile != null) ? changelogFile.lastModified() : null;
> ```

I think that should have worked fine, where was the NPE being thrown?

> was instead putting Long changelogOriginalModifiedDate = null; just above the beginning of the try... block.

That's fine too, that's probably what I would do, and then you can just set `changelogOriginalModifiedDate = changelogFile.lastModified()` inside of the `if (changelog) {` block right at the end where you and SpotBugs know `changelogFile` is non-null.

Reply:
I ended up trying it both ways :)  Settled on what I checked in. It does work. Between distractions I did what you suggested (setting it inside the very end of that `if`). And you called it on Spotbugs, I went rounds with it on Friday.

NPE was being thrown where I'd left something silly behind that was comparing the dates and checking to make sure that the changelog file was being created.

Going to add a test as well.